File: 0485-max-consecutive-ones/0485-max-consecutive-ones.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        l = 0
        r = 0
        max_len = 0
        
        while r < len(nums):
            if nums[r] == 1:
                max_len = max(max_len, r-l+1)
                r += 1
            else:
                l = r+1
                r +=1
        return max_len

File: 0485-max-consecutive-ones/0485-max-consecutive-ones.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums)
{
    int l = 0, max = 0, r = 0;
    
    while(r < nums.length)
    {
        if(nums[r] == 1)
        {
            max = Math.max(max,r-l+1);
            r++;
        }
        else
        {
            l = r + 1;
            r = l;
            
        }
        
    }
    return max;
}
}